webapp/views.py

```diff
@@ -393,6 +394,44 @@ def get_notice(notice_id, **kwargs):
             404,
         )
 
+    releases = {
+        release.codename: release.version
+        for release in db_session.query(Release).all()
+    }
+
+    package_descriptions = {}
+    package_versions = {}
+    release_packages = SortedDict()
+
+    if notice.release_packages:
+        for codename, pkgs in notice.release_packages.items():
+            release_version = releases[codename]
+            release_packages[release_version] = {}
+            for package in pkgs:
+                if not package.get("is_visible", True):
```

Comment:
What I'm going to say here is *not* super important. Of course code gets complex, and you can't always write everything perfectly. But it seems like an opportunity for me to illustrate some things I care about in writing accessible code, which I don't know whether you already know / thought about or not. So feel free to ignore this if you want, I just wanted to get it down.

So. This is clever, but a bit hard to follow:

- If `is_visible` exists, apply `not`, so `False` (invisible) will hit `continue` (skip to next item in the loop), `True` will continue down this function
- If `is_visible` doesn't exist, return `True` instead, then apply `not` to get `False`, skip over `continue` to continue down the function.

Three ways I can think of to make this a little easier to follow for people:

- `package.get("is_visible", True)` can also be written as `package.get("is_visible") or True`, which I find *slightly* more straightforward to read as you don't need to know that the second argument to the `get` method is the fallback default.
- Do the `visible = package.get("is_visible") or True` first, so then you can do `if not visible:`. This avoids one extra thought loop when trying to understand the conditional.
- Consider inverting the conditional by instead of `if not visible: continue`, you could wrap *all* the following code in `if visible:`. This of course means the code gets super indented, but on the other hand you avoid the confusion of `continue` kind meaning the exact opposite...
- Add a comment (probably not necessary if we did all of the above)

I'm not trying to say my answers here are right, just that they're options, and it's worth considering how hard it will be for people to follow your logical structures.